hippunfold/workflow/scripts/create_warps.py

```diff
@@ -104,74 +97,63 @@ def summary(name, array):
 
 # scattered interpolation / griddata:
 
-# matlab:  interp_X = scatteredInterpolant(Laplace_AP,Laplace_PD,Laplace_IO,native_coords_phys(:,1),interp,extrap);
-
 # we have points defined by coord_flat_{ap,pd,io}, and corresponding value as native_coords_phys[:,i]
 # and we want to interpolate on a grid in the unfolded space
 
-# add some noise to avoid perfectly overlapping datapoints!
-points = (
-    coord_flat_ap + (np.random.rand(coord_flat_ap.shape[0]) - 0.5) * 1e-6,
-    coord_flat_pd + (np.random.rand(coord_flat_ap.shape[0]) - 0.5) * 1e-6,
-    coord_flat_io + (np.random.rand(coord_flat_ap.shape[0]) - 0.5) * 1e-6,
-)
+# unnormalize and add a bit of noise so points don't ever perfectly overlap
+coord_flat_ap_unnorm = coord_flat_ap * unfold_dims[0]
+coord_flat_pd_unnorm = coord_flat_pd * unfold_dims[1]
+coord_flat_io_unnorm = coord_flat_io * unfold_dims[2]
 
 # get unfolded grid (from 0 to 1, not world coords), using meshgrid:
 #  note: indexing='ij' to swap the ordering of x and y
 epsilon = snakemake.params.epsilon
 (unfold_gx, unfold_gy, unfold_gz) = np.meshgrid(
-    np.linspace(0 + float(epsilon[0]), 1 - float(epsilon[0]), unfold_dims[0]),
-    np.linspace(0 + float(epsilon[1]), 1 - float(epsilon[1]), unfold_dims[1]),
-    np.linspace(0 + float(epsilon[2]), 1 - float(epsilon[2]), unfold_dims[2]),
+    np.linspace(
+        0 + float(epsilon[0]), unfold_dims[0] - float(epsilon[0]), unfold_dims[0]
+    ),
+    np.linspace(
+        0 + float(epsilon[1]), unfold_dims[1] - float(epsilon[1]), unfold_dims[1]
+    ),
+    np.linspace(
+        0 + float(epsilon[2]), unfold_dims[2] - float(epsilon[2]), unfold_dims[2]
+    ),
     indexing="ij",
 )
-
-# tuple for use in griddata:
-unfold_xi = (unfold_gx, unfold_gy, unfold_gz)
 summary("unfold_gx", unfold_gx)
 summary("unfold_gy", unfold_gy)
 summary("unfold_gz", unfold_gz)
 
-# perform the interpolation, filling in outside values as 0
-#  TODO: linear vs cubic?  we were using "natural" interpolation in matlab
-#         so far, linear seems close enough..
-interp_ap = griddata(
-    points, values=native_coords_phys[:, 0], xi=unfold_xi, method=interp_method
-)
-summary("interp_ap", interp_ap)
-interp_pd = griddata(
-    points, values=native_coords_phys[:, 1], xi=unfold_xi, method=interp_method
-)
-summary("interp_pd", interp_pd)
-interp_io = griddata(
-    points, values=native_coords_phys[:, 2], xi=unfold_xi, method=interp_method
+# perform the interpolation
+
```

Comment:
actually looks like noise is added here - so just need to move the comment

Reply:
Actually, `unfolded_gx` `unfolded_gy` `unfolded_gz` were no longer used, so I removed them. This also means that epsilon is no longer used, and i believe its no longer needed